### DISCUSSION TITLE
Scan does not respect sha pined versions and fall back to the last tagged version

### DISCUSSION THREAD

**Discussion by 6543** [aquasecurity/trivy#7430] [Bugs]
### Description

Trivy detected an issue with an golang lib: https://ci.woodpecker-ci.org/repos/3780/pipeline/20094/19
so i created a pull to fix it: https://github.com/woodpecker-ci/woodpecker/pull/4077

but trivy now shows me an false positive: https://ci.woodpecker-ci.org/repos/3780/pipeline/20098/16

### Desired Behavior

It would be nice if trivy could just parse it as the next version ... or have a config that let the case specify if it should in this case use the next version or as of now use the last tag

### Actual Behavior

as described above

### Reproduction Steps

```bash
use golang and pin version by getting it via an hash (e.g. `go get github.com/moby/moby@852759a7df454cbf88db4e954c919becd
48faa9b`)
```

### Target

Filesystem

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
irrelevant
```

### Operating System

linux x86

### Version

```bash
Version: 0.52.0 (woodpeckerci/plugin-trivy:1.1.0)
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
As written in [the advisory](https://github.com/advisories/GHSA-v23v-6jw2-98fq), v24.x.y is considered vulnerable. I suppose that's because [the patch](https://github.com/moby/moby/commit/852759a7df454cbf88db4e954c919becd48faa9b) is not yet released. Since Trivy basically follows security advisories, it's hard for Trivy to handle this case. You can ask Docker to cut v0.24.0.10 and update the advisory.

Until then, you can suppress the vulnerability.
https://aquasecurity.github.io/trivy/v0.54/docs/configuration/filtering/#suppression
